AP_Motors: always use PWM min and max, don't take from throttle.

IamPete1 opened the pull request:
Fixes #13421 #10488

On copter and sub this removes usage of RC input PWM range as the range for motor outputs. Currently if motors PWM min/max are not set changing your RC calibration will change the output ranges.

On plane this removes the duplicate params `QTHR_MIN_PWM` and `QTHR_MAX_PWM`

amilcarlucas commented:
This looks like a @peterbarker PR. It removes more lines than it adds! :)

rmackay9 commented:
well, this is nice.. we shouldn't let this one get away..

IamPete1 commented on ArduPlane/quadplane.cpp:
```diff
@@ -545,6 +530,9 @@ const AP_Param::ConversionInfo q_conversion_table[] = {
     { Parameters::k_param_quadplane, 1659,  AP_PARAM_FLOAT, "Q_TAILSIT_RAT_FW" },
+
+    { Parameters::k_param_quadplane, 22,  AP_PARAM_FLOAT, "Q_M_PWM_MIN" },
+    { Parameters::k_param_quadplane, 23,  AP_PARAM_FLOAT, "Q_M_PWM_MAX" },
```
```suggestion
    { Parameters::k_param_quadplane, 23,  AP_Int16, "Q_M_PWM_MAX" },
```

IamPete1 replied:
yep, this did not work. fixed.

tridge commented:
need to make sure that a new user that does R/C cal first doesn't end up with MOT_PWM_MIN/MAX set to RC min/max

IamPete1 commented on libraries/AP_Motors/AP_MotorsMulticopter.cpp:
```diff
@@ -459,59 +456,23 @@ float AP_MotorsMulticopter::actuator_spin_up_to_ground_idle() const
     }
     // negative values are out-of-range:
-    if (_pwm_min < 0 || _pwm_max < 0) {
+    if (_pwm_max < 0) {
```
```suggestion
    if (_pwm_max <= 0) {
```

IamPete1 commented:
Fixed the plane param conversion. 

Updated the copter param conversion, @rmackay9  I think this now works as you had in mind. 

Were a bit stuck on the sub param conversion, because we have never been using MOT_PWM_MIN and _MAX they might be configured to some nonsense value that we would now start using. If they have never been set the conversion will work fine.

IamPete1 commented:
We do have a arming check for PWM min/max, but were only calling on copter. We could add that check to Qplane and Sub too.

rmackay9 commented on libraries/AP_Motors/AP_MotorsMulticopter.h:
```diff
@@ -44,9 +44,7 @@ class AP_MotorsMulticopter : public AP_Motors {
-    // set_throttle_range - sets the minimum throttle that will be sent to the engines when they're not off (i.e. to prevents issues with some motors spinning and some not at very low throttle)
-    // also sets minimum and maximum pwm values that will be sent to the motors
-    void                set_throttle_range(int16_t radio_min, int16_t radio_max);
+    void                update_throttle_range();
```
ideally we should put a comment above this method.

Williangalvani commented:
This seems to be changing the defaults from 1100-1900 to 1000-2000. Is that on purpose?

IamPete1 commented:
@Williangalvani  Yes, the change in defaults is by design, we can `#define` it if you don't want that on sub.

Williangalvani reviewed (approved):
Sub-wise this looks good. Thanks for fixing the issue!

rmackay9 commented:
Merged, thanks!
